### PR TITLE
Android: Allow configure params

### PR DESCRIPTION
**Pull request by m1ga**
Allowing to pass parameters into the configure() method.

* APIKey
* databaseUrl
* projectId
* storageBucket
* applicationId
* gcmSenderId

Names are from https://firebase.google.com/docs/reference/android/com/google/firebase/FirebaseOptions
Couldn't match all the iOS names

### CODE REVIEW

File: android/src/firebase/core/TitaniumFirebaseCoreModule.java [hunk 1]
```diff
@@ -37,9 +39,34 @@ public static void onAppCreate(TiApplication app)
...
+			if (param.containsKey("storageBucket")) {
+				options.setStorageBucket(param.getString("storageBucket"));
+			}
+			if (param.containsKey("applicationId")) {
+				options.setApplicationId(param.getString("applicationId"));
+			}
+			if (param.containsKey("gcmSenderId")) {
+				options.setGcmSenderId(param.getString("gcmSenderId"));
+			}
+			if (param.containsKey("gcmSenderId")) {
```
Comment (hansemannn) [inline android/src/firebase/core/TitaniumFirebaseCoreModule.java]: Duplicated key from line 61?

File: android/src/firebase/core/TitaniumFirebaseCoreModule.java [hunk 2]
```diff
@@ -37,9 +39,34 @@ public static void onAppCreate(TiApplication app)
...
-    // TODO: Allow passing options as well. Right now, it will use the default Firebase-instance
-    FirebaseApp.initializeApp(getActivity().getApplicationContext());
-  }
+		if (param != null) {
+			FirebaseOptions.Builder options = new FirebaseOptions.Builder();
+			if (param.containsKey("APIKey")) {
+				options.setApiKey(param.getString("APIKey"));
+			}
+			if (param.containsKey("databaseUrl")) {
+				options.setDatabaseUrl(param.getString("databaseUrl"));
```
Comment (hansemannn) [inline android/src/firebase/core/TitaniumFirebaseCoreModule.java]: Please call `databaseURL`.

File: android/src/firebase/core/TitaniumFirebaseCoreModule.java [hunk 3]
```diff
@@ -37,9 +39,34 @@ public static void onAppCreate(TiApplication app)
...
+			}
+			if (param.containsKey("projectId")) {
```
Comment (hansemannn) [inline android/src/firebase/core/TitaniumFirebaseCoreModule.java]: Please call `projectID`.

File: android/src/firebase/core/TitaniumFirebaseCoreModule.java [hunk 4]
```diff
@@ -37,9 +39,34 @@ public static void onAppCreate(TiApplication app)
...
+				options.setProjectId(param.getString("projectId"));
+			}
+			if (param.containsKey("storageBucket")) {
+				options.setStorageBucket(param.getString("storageBucket"));
+			}
+			if (param.containsKey("applicationId")) {
+				options.setApplicationId(param.getString("applicationId"));
+			}
+			if (param.containsKey("gcmSenderId")) {
```
Comment (hansemannn) [inline android/src/firebase/core/TitaniumFirebaseCoreModule.java]: Please call `GCMSenderID`.

File: android/src/firebase/core/TitaniumFirebaseCoreModule.java [hunk 5]
```diff
@@ -37,9 +39,34 @@ public static void onAppCreate(TiApplication app)
...
+			if (param.containsKey("databaseUrl")) {
+				options.setDatabaseUrl(param.getString("databaseUrl"));
+			}
+			if (param.containsKey("projectId")) {
+				options.setProjectId(param.getString("projectId"));
+			}
+			if (param.containsKey("storageBucket")) {
+				options.setStorageBucket(param.getString("storageBucket"));
+			}
+			if (param.containsKey("applicationId")) {
```
Comment (hansemannn) [inline android/src/firebase/core/TitaniumFirebaseCoreModule.java]: I think it's the same as the `bundleID` used for iOS, so we good get good parity here if you can rename it. From the Firebase Android docs:
```
android {
    defaultConfig {
        applicationId "com.example.myapp"
        minSdkVersion 15
        targetSdkVersion 24
        versionCode 1
        versionName "1.0"
    }
    ...
}
```
So setting the `bundleID` to `Ti.App.id` could set it for both platforms.

**Review by hansemannn — CHANGES_REQUESTED** [review]
Added some parity recommendations. Please also update the readme if possible. We could do a table with properties and checkmark the supported platforms.

**Comment by m1ga** [comment]
about the applicationId. I've found this example:
```
FirebaseOptions.Builder builder = new FirebaseOptions.Builder()
    .setApplicationId("1:0123456789012:android:0123456789abcdef")
    .setApiKey("your_api_key")
    .setDatabaseUrl("https://your-app.firebaseio.com")
    .setStorageBucket("your-app.appspot.com");
```
which is not using the bundleid as the applicationid. It might be different

Ok it is diffrent:

_The term "application ID" unfortunately can refer to two different things.  To initialize Firebase, don't use the android package name.  The package name of your can be found by the SDK at runtime using other methods._

_The application ID you want to use to initialize Firebase is stored in google-service.json under client/client_info/mobilesdk_app_id.  It probably starts with "1:" and ends with some hex numbers.  It's the same "App ID" you see for your app in the Firebase Console when you look at it under project settings._
https://groups.google.com/forum/#!topic/firebase-talk/cLhcPdVv2es

**Comment by hansemannn** [comment]
Sad that Google does not care about more parity here. Then let's take this one and document the parity discrepancies.

**Review by hansemannn — APPROVED** [review]

**Comment by m1ga** [comment]
updated the documentation!

**Comment by m1ga** [comment]
BTW: https://firebase.google.com/docs/configure/#use_multiple_projects_in_your_application
there is an example where you can switch between iOS/Android to see the differences. Strange that they did it like this

**Comment by hansemannn** [comment]
New version is released! Also added 64 Bit support in [Android 2.0.0](https://github.com/hansemannn/titanium-firebase-core/releases/tag/android-2.0.0). Good times! 🙂

**Pull request merged**